### PR TITLE
Perf tests no longer depend on integration tests

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
The infrastructure necessary to run performance tests on both the
in-memory and the persistent sandbox is duplicated to the
//ledger/sandbox-perf package in order to be able to get rid of all
contents of //ledger/ledger-api-integration-tests once the test porting
is over (which will happen soon).

The scope and functionality of ported classes has been reduced to the
minimum set needed by the performance test suite.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags, if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by SamirTalwar** [comment]
This mostly looks good. I am wary that the perf tests seem to have passed in 3 minutes; typically they take around 12. Unfortunately, the logs aren't loading for me, so I can't check what's going on. 🙁

**Comment by SamirTalwar** [comment]
Please double-check that everything is actually running before merging. I've tried diving into the logs but can't get very far.

File: ledger/sandbox-perf/src/perf/lib/scala/com/digitalasset/platform/sandbox/perf/LedgerContext.scala [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import scalaz.syntax.tag._
+
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+
+final class LedgerContext(channel: Channel, packageIds: Iterable[PackageId])(
+    implicit protected val mat: ActorMaterializer,
+    implicit protected val esf: ExecutionSequencerFactory) {
+
+  require(esf != null, "ExecutionSequencerFactory must not be null.")
```
[the remark this answers is not in the document]
Comment (stefanobaghino-da) [reply ledger/sandbox-perf/src/perf/lib/scala/com/digitalasset/platform/sandbox/perf/LedgerContext.scala]: IDK YOLO let's find out.

File: ledger/sandbox-perf/src/perf/lib/scala/com/digitalasset/platform/sandbox/perf/LedgerFactories.scala [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      authService = None,
+    )
+
+  val mem = "InMemory"
+  val sql = "Postgres"
+
+  def createSandboxResource(store: String, darFiles: List[File])(
+      implicit esf: ExecutionSequencerFactory,
+      mat: ActorMaterializer): Resource[LedgerContext] = {
+    val packageIds = darFiles.map(getPackageIdOrThrow)
```
[the remark this answers is not in the document]
Comment (stefanobaghino-da) [reply ledger/sandbox-perf/src/perf/lib/scala/com/digitalasset/platform/sandbox/perf/LedgerFactories.scala]: Addressed by https://github.com/digital-asset/daml/pull/3842/commits/7140b764866192740def4af1edb4dbcf38d81997

File: ledger/sandbox-perf/src/perf/lib/scala/com/digitalasset/platform/sandbox/perf/LedgerFactories.scala [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+object LedgerFactories {
+
+  private def getPackageIdOrThrow(file: File): Ref.PackageId =
+    UniversalArchiveReader().readFile(file).map(_.all.head._1).get
+
+  private def sandboxConfig(jdbcUrl: Option[String], darFiles: List[File]) =
+    SandboxConfig(
+      address = None,
+      port = 0,
+      None,
```
[the remark this answers is not in the document]
Comment (stefanobaghino-da) [reply ledger/sandbox-perf/src/perf/lib/scala/com/digitalasset/platform/sandbox/perf/LedgerFactories.scala]: Addressed by https://github.com/digital-asset/daml/pull/3842/commits/e6fd4cfdc9e65fa650778ae358a960ac4c9c0a7f

File: ledger/sandbox-perf/src/perf/lib/scala/com/digitalasset/platform/sandbox/perf/LedgerContext.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (stefanobaghino-da) [reply ledger/sandbox-perf/src/perf/lib/scala/com/digitalasset/platform/sandbox/perf/LedgerContext.scala]: Addressed by https://github.com/digital-asset/daml/pull/3842/commits/d580b3d9842691d2529c00bd1be55aa2d402f2a2

**Pull request merged**